pyaedt/edb_core/edb_data/simulation_setup.py

```diff
@@ -0,0 +1,701 @@
+
+    @name.setter
+    def name(self, value):
+        self._pedb.layout.cell.DeleteSimulationSetup(self.name)
+        self._edb_object.Name = value
+        self._name = value
+        self._update_setup()
+
+    @property
+    def position(self):
+        """Position in the setup list."""
+        return self._edb_object.Position
+
+    @position.setter
+    def position(self, value):
+        self._edb_object.Position = value
+
+    @property
+    def setup_type(self):
+        """Type of the setup."""
+        return self._edb_object.SimSetupType.ToString()
+
+    @property
+    def frequency_sweeps(self):
+        """list of frequency sweep."""
+        temp = {}
+        for i in list(self._edb_object.SweepDataList):
+            temp[i.Name] = EdbFrequencySweep(self, None, i.Name, i)
+        return temp
+
+    @pyaedt_function_handler
+    def _create(self, name=None):
+        """Create a new setup."""
+        if not name:
+            name = generate_unique_name(self.setup_type)
+            self._name = name
+
+        if self._edb_object:
+            self._setup_type = self.setup_type
+
+        self._edb_object = self._pedb.simsetupdata.SimSetupInfo[self._setup_type_mapping[self._setup_type]]()
+        self._edb_object.Name = name
+        self._update_setup()
+
+    @pyaedt_function_handler
+    def _generate_edb_setup(self):
+        """Generate convert simulation setup information into the format Edb can import."""
+        setup_type_mapping = {
+            "kHFSS": self._pedb.edb_api.utility.utility.HFSSSimulationSetup,
+            "kPEM": None,
+            "kSIwave": self._pedb.edb_api.utility.utility.SIWaveSimulationSetup,
+            "kLNA": None,
+            "kTransient": None,
+            "kQEye": None,
+            "kVEye": None,
+            "kAMI": None,
+            "kAnalysisOption": None,
+            "kSIwaveDCIR": self._pedb.edb_api.utility.utility.SIWaveDCIRSimulationSetup,
+            "kSIwaveEMI": None,
+            "kHFSSPI": None,
+            "kDDRwizard": None,
+            "kQ3D": None,
+            "kNumSetupTypes": None,
+        }
+        return setup_type_mapping[self.setup_type](self._edb_object)
+
+    @pyaedt_function_handler()
+    def _update_setup(self):
+        """Update setup into Edb."""
+        if self._setup_type == "kHFSS":
+            mesh_operations = self._edb_object.SimulationSettings.MeshOperations
+            mesh_operations.Clear()
+            for mop in self.mesh_operations.values():
+                mesh_operations.Add(mop.mesh_operation)
+
+        edb_setup = self._generate_edb_setup()
+        if self.name in self._pedb.setups:
+            self._pedb.layout.cell.DeleteSimulationSetup(self.name)
+        self._pedb.layout.cell.AddSimulationSetup(edb_setup)
+        return True
+
+
+class EdbFrequencySweep(object):
+    """Manages EDB methods for frequency sweep."""
+
+    def __init__(self, sim_setup, frequency_sweep=None, name=None, edb_sweep_data=None):
```

Comment:
The class should have a Parameters section with all the parameters defined in the __init__ method defined.